### PR TITLE
add vulkan-validationlayers/1.2.154.0

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **vulkan-validationlayers/1.2.154.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

closes https://github.com/conan-io/conan-center-index/issues/1347

### CODE REVIEW

**Review by madebr — DISMISSED** [review]
Does this recipe depend on the vulkan-loader pr?

**Comment by SpaceIm** [comment]
No, but I need to create a simple test for util library (allow to create validation layers) and khronos validation layer (the module).

**Review by prince-chrismc — DISMISSED** [review]

**Review by madebr — DISMISSED** [review]

File: recipes/vulkan-validationlayers/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            del self.options.with_wsi_wayland
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 11)
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+            raise ConanInvalidConfiguration("gcc < 5 is not supported")
+
+    def requirements(self):
+        self.requires("glslang/8.13.3559")
```
Comment (SpaceIm) [inline recipes/vulkan-validationlayers/all/conanfile.py:51]: Actually, it seems that glslang is not required (only for tests). I'll try some tricks to make upstream CMakeLists happy with a fake glslang interface target.

Comment (SpaceIm) [reply recipes/vulkan-validationlayers/all/conanfile.py]: Moreover, spirv-headers is also a direct dependency of vulkan-validationlayers, but:
- spirv-tools also depends on spirv-headers
- spirv-tools is very strict on spirv-headers version, so I would prefer to not introduce conflicts.

File: recipes/vulkan-validationlayers/all/CMakeLists.txt [hunk 2]
```diff
@@ -4,10 +4,11 @@ project(cmake_wrapper)
 include(conanbuildinfo.cmake)
 conan_basic_setup(TARGETS)
 
-add_library(glslang INTERFACE)
-target_link_libraries(glslang INTERFACE CONAN_PKG::glslang)
+add_library(glslang INTERFACE) # fake target for upstream CMakeLists (glslang required by tests only)
```
Comment (prince-chrismc) [inline recipes/vulkan-validationlayers/all/CMakeLists.txt:7]: I really like this more then patches

**Review by prince-chrismc — DISMISSED** [review]

**Comment by SpaceIm** [comment]
Last change: spirv-tools can't be shared, because SPIRV-Tools-opt is required (not built when spirv-tools is shared). It also means that spirv-tools can be considered as private in the sense of conan (the plugin depends on spirv-tools, but not the static lib).

**Review by prince-chrismc — APPROVED** [review]

File: recipes/vulkan-validationlayers/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+required_conan_version = ">=1.32.0"
+
+
+class VulkanValidationLayersConan(ConanFile):
+    name = "vulkan-validationlayers"
+    description = "Khronos official Vulkan validation layers for Windows, Linux, Android, and MacOS."
+    license = "Apache-2.0"
+    topics = ("conan", "vulkan-validation-layers", "vulkan", "validation-layers")
+    homepage = "https://github.com/KhronosGroup/Vulkan-ValidationLayers"
```
Comment (madebr) [inline recipes/vulkan-validationlayers/all/conanfile.py:15]: Small remark before collecting the final approvals, how about naming this recipe `vulkan-validation-layers`?
The 'L' in of `Layers` in `ValidationLayers` is a capital.

Comment (SpaceIm) [reply recipes/vulkan-validationlayers/all/conanfile.py:15]: there is no clear consensus here: https://repology.org/project/vulkan-validationlayers/versions

Comment (madebr) [reply recipes/vulkan-validationlayers/all/conanfile.py:15]: Thanks for the link!
I won't push further.

**Review by madebr — APPROVED** [review]

File: recipes/vulkan-validationlayers/all/CMakeLists.txt (hunk 2 above)
Comment (uilianries) [reply recipes/vulkan-validationlayers/all/CMakeLists.txt:7]: At least, it works for any version. Patches are pretty limited

**Review by uilianries — APPROVED** [review]

**Pull request merged**